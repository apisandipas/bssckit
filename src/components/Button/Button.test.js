import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import { Button, LinkButton } from './index';

test('Styles matches Button', () => {
  const { container } = render(<Button />);

  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('display', 'inline-block');
  expect(container.firstChild).toHaveStyleRule('font-weight', '400');
  expect(container.firstChild).toHaveStyleRule('text-align', 'center');
  expect(container.firstChild).toHaveStyleRule('white-space', 'nowrap');
  expect(container.firstChild).toHaveStyleRule('vertical-align', 'middle');
  expect(container.firstChild).toHaveStyleRule('user-select', 'none');
  expect(container.firstChild).toHaveStyleRule('border', '1px solid #6c757d');
  expect(container.firstChild).toHaveStyleRule('padding', '0.375rem 0.75rem');
  expect(container.firstChild).toHaveStyleRule('font-size', '1rem');
  expect(container.firstChild).toHaveStyleRule('line-height', '1.5');
  expect(container.firstChild).toHaveStyleRule('border-radius', '0.25rem');
  expect(container.firstChild).toHaveStyleRule(
    'transition',
    'color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out'
  );
  expect(container.firstChild).toHaveStyleRule('text-decoration', 'none', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('text-decoration', 'none', {
    modifier: '&:focus',
  });
  expect(container.firstChild).toHaveStyleRule('outline', '0', {
    modifier: '&:focus',
  });
  expect(container.firstChild).toHaveStyleRule('box-shadow', '0 0 0 0.2rem rgba(108,117,125,0.5)', {
    modifier: '&:focus',
  });
});

test('Styles match LinkButton', () => {
  const { container } = render(<LinkButton />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Styles match Button active', () => {
  const { container } = render(<Button active />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('background-image', 'none');
});

test('Styles match Button block', () => {
  const { container } = render(<Button block />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('display', 'block');
  expect(container.firstChild).toHaveStyleRule('width', '100%');
});

test('Styles match Button disabled', () => {
  const { container } = render(<Button disabled />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('opacity', '0.65');
});

test('Styles matches Button primary disabled', () => {
  const { container } = render(<Button primary disabled />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('color', '#fff');
  expect(container.firstChild).toHaveStyleRule('background-color', '#007bff');
  expect(container.firstChild).toHaveStyleRule('border', '1px solid #007bff');
  expect(container.firstChild).toHaveStyleRule('color', '#fff', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('background-color', '#007bff', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('border', '1px solid #007bff', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('box-shadow', '0 0 0 0.2rem rgba(0,123,255,0.5)', {
    modifier: '&:focus',
  });
});

test('Styles match Button primary', () => {
  const { container } = render(<Button primary />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Styles match Button primary active', () => {
  const { container } = render(<Button primary active />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Styles match Button primary outline disabled', () => {
  const { container } = render(<Button primary outline disabled />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('color', '#007bff');
  expect(container.firstChild).toHaveStyleRule('background-color', 'transparent');
  expect(container.firstChild).toHaveStyleRule('background-image', 'none');
  expect(container.firstChild).toHaveStyleRule('border', '1px solid #007bff');
  expect(container.firstChild).toHaveStyleRule('color', '#007bff', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('background-color', 'transparent', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('background-image', 'none', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('border', '1px solid #007bff', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('box-shadow', '0 0 0 0.2rem rgba(0,123,255,0.5)', {
    modifier: '&:focus',
  });
});

test('Styles match Button primary outline', () => {
  const { container } = render(<Button primary outline />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveStyleRule('color', '#007bff');
  expect(container.firstChild).toHaveStyleRule('background-color', 'transparent');
  expect(container.firstChild).toHaveStyleRule('background-image', 'none');
  expect(container.firstChild).toHaveStyleRule('border', '1px solid #007bff');
  expect(container.firstChild).toHaveStyleRule('color', '#fff', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('background-color', '#007bff', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('border', '1px solid #007bff', {
    modifier: '&:hover',
  });
  expect(container.firstChild).toHaveStyleRule('box-shadow', '0 0 0 0.2rem rgba(0,123,255,0.5)', {
    modifier: '&:focus',
  });
});
