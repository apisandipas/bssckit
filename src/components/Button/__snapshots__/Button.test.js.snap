// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles match Button active 1`] = `
.c0 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  background-image: none;
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

<button
  class="sc-AxiKw c0"
/>
`;

exports[`Styles match Button block 1`] = `
.c1 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  cursor: pointer;
  display: block;
  width: 100%;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
}

.c1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

.c1 + .c0 {
  margin-top: 0.5rem;
}

<button
  class="sc-AxiKw c0 c1"
/>
`;

exports[`Styles match Button disabled 1`] = `
.c0 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  opacity: 0.65;
  background-image: none;
  cursor: not-allowed;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

<button
  class="sc-AxiKw c0"
  disabled=""
/>
`;

exports[`Styles match Button primary 1`] = `
.c0 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
  color: #fff;
  border: 1px solid #007bff;
  background-color: #007bff;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

.c0:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0,123,255,0.5);
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  border: 1px solid #007bff;
  background-color: #007bff;
}

.c0:hover:focus,
.c0:focus:focus,
.c0:hover:hover,
.c0:focus:hover {
  border: 1px solid #005cbf;
}

.c0:hover:focus,
.c0:focus:focus,
.c0:hover:hover,
.c0:focus:hover {
  background-color: #0069d9;
}

.c0:focus,
.c0:hover {
  border: 1px solid #005cbf;
}

.c0:focus,
.c0:hover {
  background-color: #0069d9;
}

<button
  class="sc-AxiKw c0"
/>
`;

exports[`Styles match Button primary active 1`] = `
.c0 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  background-image: none;
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
  color: #fff;
  border: 1px solid #005cbf;
  background-image: none;
  background-color: #005cbf;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

.c0:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0,123,255,0.5);
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  border: 1px solid #005cbf;
  background-image: none;
  background-color: #005cbf;
}

<button
  class="sc-AxiKw c0"
/>
`;

exports[`Styles match Button primary outline 1`] = `
.c0 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
  color: #007bff;
  border: 1px solid #007bff;
  background-image: none;
  background-color: transparent;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

.c0:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0,123,255,0.5);
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #007bff;
  border: 1px solid #007bff;
  background-image: none;
  background-color: transparent;
}

.c0:hover:hover,
.c0:focus:hover {
  color: #fff;
}

.c0:hover:hover,
.c0:focus:hover {
  background-color: #007bff;
}

.c0:hover {
  color: #fff;
}

.c0:hover {
  background-color: #007bff;
}

<button
  class="sc-AxiKw c0"
/>
`;

exports[`Styles match Button primary outline disabled 1`] = `
.c0 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  opacity: 0.65;
  background-image: none;
  cursor: not-allowed;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
  border: 1px solid #007bff;
  color: #007bff;
  border: 1px solid #007bff;
  background-image: none;
  background-color: transparent;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

.c0:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0,123,255,0.5);
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #007bff;
  border: 1px solid #007bff;
  background-image: none;
  background-color: transparent;
}

.c0:hover:hover,
.c0:focus:hover {
  background-color: transparent;
}

.c0:hover {
  background-color: transparent;
}

<button
  class="sc-AxiKw c0"
  disabled=""
/>
`;

exports[`Styles match LinkButton 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

<a
  class="sc-AxhCb c0"
/>
`;

exports[`Styles matches Button 1`] = `
.c0 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

<button
  class="sc-AxiKw c0"
/>
`;

exports[`Styles matches Button primary disabled 1`] = `
.c0 {
  font-weight: 400;
  line-height: 1.5;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  border: 1px solid #6c757d;
  opacity: 0.65;
  background-image: none;
  cursor: not-allowed;
  display: inline-block;
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
  border-radius: 0.25rem;
  border: 1px solid #007bff;
  color: #fff;
  border: 1px solid #007bff;
  background-image: none;
  background-color: #007bff;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108,117,125,0.5);
}

.c0:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0,123,255,0.5);
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  border: 1px solid #007bff;
  background-image: none;
  background-color: #007bff;
}

<button
  class="sc-AxiKw c0"
  disabled=""
/>
`;
